server/src/actors/project.rs

```diff
@@ -518,9 +606,26 @@ impl Handler<GetEventAction> for Project {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
 pub struct RetryAfter {
-    pub secs: u64,
+    pub when: Instant,
+    pub reason_code: Option<String>,
```

Comment:
👏 

Note that in https://github.com/getsentry/semaphore/pull/236 `reason_code` is non-optional.